core/store/src/flat_state.rs

```diff
@@ -46,35 +146,33 @@ mod imp {
         /// are stored in `DBCol::State`. Also the separation is done so we
         /// could charge users for the value length before loading the value.
         // TODO (#7327): support different roots (or block hashes).
-        pub fn get_ref(
-            &self,
-            _root: &CryptoHash,
-            key: &[u8],
-        ) -> Result<Option<ValueRef>, StorageError> {
-            match self.get_raw_ref(key)? {
+        pub fn get_ref(&self, key: &[u8]) -> Result<Option<ValueRef>, StorageError> {
```

Comment:
If you want to implement this function properly, can you also implement https://nearinc.atlassian.net/browse/STR-18 in this PR?

Reply:
Yes, in fact, it is already implemented here.